### PR TITLE
CI: Freeze Mycroft core version

### PR DESCRIPTION
**Pull request by Tony763**
#### Description
Hi, this PR adds variable with Mycroft Core version to freeze it. This should stabilize CI environment as current Mycroft core dev branch seems to have problems with installation.

#### Type of PR
- [x] Bugfix
- [ ] Feature implementation
- [ ] Refactor of code (without functional changes)
- [ ] Documentation improvements
- [ ] Test improvements

### CODE REVIEW

**Comment by Tony763** [comment]
Hi @krisgesling, I know you have hands full with MK2 production, but this one should be fast to review.

PS: Hacktoberfest is here, so if accepted, add please a label.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -84,10 +85,10 @@ jobs:
       - name: Mycroft - install Mycroft
         run: |
           sudo apt update
-          git clone https://github.com/MycroftAI/mycroft-core.git /opt/mycroft-core
+          git clone --depth 1 --branch ${{ env.MYCROFT_CORE_BRANCH }} https://github.com/MycroftAI/mycroft-core.git /opt/mycroft-core
           ls /opt
           chmod +x /opt/mycroft-core/dev_setup.sh
-          /opt/mycroft-core/dev_setup.sh --allow-root -ci -sm
```
Comment (krisgesling) [inline .github/workflows/build.yml:90]: should the `-ci` flag remain so the setup process doesn't prompt with questions?

Comment (Tony763) [reply .github/workflows/build.yml:90]: No, arg `-ci` was used in the past when I did some changes to `dev_setu.sh` to run in CI. In latest released version of it, setup look for `$CI`  environment variable, so questions are handled automatically.

ref: https://github.com/MycroftAI/mycroft-core/blob/5f964f7d4036210c7e1977e6903d3f13421e2af2/dev_setup.sh#L175

Comment (Tony763) [reply .github/workflows/build.yml:90]: Working workflow  can be seen [there](https://github.com/Tony763/skill-homeassistant/actions/runs/3240230182/jobs/5310605787)

**Comment by Tony763** [comment]
Hi @krisgesling, could You please merge this?

**Comment by stratus-ss** [comment]
I think this is ok to merge, but I am not exactly clear what the implications of this are on the current process(es). @krisgesling is there something that needs to be updated elsewhere to take advantage of this change?

**Comment by Tony763** [comment]
Hi @stratus-ss, this should have no big impact. This just freeze a version of Mycroft Core used in CI. Some last changes to Core did break it and all test done by CI in this repository were/are marked as fail as Core did not install at all.

After this change there is only one drawback, once in a while, somebody from you (or me) should check if Core did not change something major and update it's version. (Something similar you do in mycroft-skills repository).

**Review by stratus-ss — APPROVED** [review]
LTGM, with Tony's explanation. I don't forsee this causing and problems going forward, aside from major version bumps in mycroft core

**Pull request merged**